typings/index.d.ts

```diff
@@ -2726,8 +2726,32 @@ export type BitFieldResolvable<T extends string, N extends number | bigint> =
 
 export type BufferResolvable = Buffer | string;
 
+export type CachedManagerTypes =
+  | 'ApplicationCommandManager'
+  | 'BaseGuildEmojiManager'
+  | 'ChannelManager'
+  | 'GuildChannelManager'
+  | 'GuildManager'
+  | 'GuildMemberManager'
+  | 'GuildBanManager'
+  | 'MessageManager'
+  | 'PermissionOverwriteManager'
+  | 'PresenceManager'
+  | 'ReactionManager'
+  | 'ReactionUserManager'
+  | 'RoleManager'
+  | 'StageInstanceManager'
+  | 'ThreadManager'
+  | 'ThreadMemberManager'
+  | 'UserManager'
+  | 'VoiceStateManager';
+
 export type CacheFactory = <T>(manager: { name: string }, holds: { name: string }) => Collection<Snowflake, T>;
```

Comment:
Might as well add types to this too.

Reply:
Done, I think it's a mess lol, but it works.